data/store/mongo/mongo.go

```diff
@@ -635,9 +635,9 @@ func (d *DataRepository) UnarchiveDeviceDataUsingHashesFromDataSet(ctx context.C
 
 	result := struct {
 		ID struct {
-			Active            bool   `bson:"_active"`
-			ArchivedDataSetID string `bson:"archivedDatasetId"`
-			ArchivedTime      string `bson:"archivedTime"`
+			Active            bool      `bson:"_active"`
+			ArchivedDataSetID string    `bson:"archivedDatasetId"`
+			ArchivedTime      time.Time `bson:"archivedTime"`
```

Comment:
Should `ArchivedTime` be of type `*time.Time` to better represent the case when `ArchivedTime` is not present in the database?